src/platform/Zephyr/DiagnosticDataProviderImpl.cpp

```diff
@@ -60,6 +64,32 @@ namespace {
 BootReasonType DetermineBootReason()
 {
 #ifdef CONFIG_HWINFO
+
+#ifdef CONFIG_SOC_SERIES_NRF54HX
+    uint32_t reason = nrf_resetinfo_resetreas_global_get(NRF_RESETINFO);
+
+    if (reason & RESETINFO_RESETREAS_GLOBAL_RESETPORONLY_Msk)
+    {
+        return BootReasonType::kBrownOutReset;
+    }
+
+    if (reason & RESETINFO_RESETREAS_GLOBAL_DOG_Msk)
```

Comment:
What about changing ifs to else ifs?

Reply:
I know this is subjective, but this is actually opposite to what `clang-tidy` considers readability improvement: https://clang.llvm.org/extra/clang-tidy/checks/readability/else-after-return.html :)